### PR TITLE
update plugins for maven 3.6.3+

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

**Review by sambish5 — APPROVED** [review]

**Comment by jpdahlke** [comment]
- [ ] validate versions are available internally

File: pom.xml [hunk 1]
```diff
@@ -87,41 +87,40 @@
...
     <maven.compiler.target>11</maven.compiler.target>
-    <maven.min-version>3.5.0</maven.min-version>
-    <plugin.build-helper-maven.version>3.3.0</plugin.build-helper-maven.version>
-    <plugin.cyclonedx-maven.version>2.7.9</plugin.cyclonedx-maven.version>
-    <plugin.duplicate-finder-maven.version>1.5.1</plugin.duplicate-finder-maven.version>
-    <plugin.exec-maven.version>3.1.0</plugin.exec-maven.version>
+    <maven.min-version>3.6.3</maven.min-version>
+    <plugin.build-helper-maven.version>3.6.0</plugin.build-helper-maven.version>
+    <plugin.cyclonedx-maven.version>2.9.0</plugin.cyclonedx-maven.version>
+    <plugin.duplicate-finder-maven.version>2.0.1</plugin.duplicate-finder-maven.version>
```
Comment (cfkoehler) [inline pom.xml:93]: not available yet

File: pom.xml [hunk 2]
```diff
@@ -87,41 +87,40 @@
...
-    <plugin.maven-scm-publish.version>3.2.1</plugin.maven-scm-publish.version>
-    <plugin.maven-site.version>3.12.1</plugin.maven-site.version>
-    <plugin.maven-source.version>3.2.1</plugin.maven-source.version>
-    <!-- seems to have an impact with java 11 integration, update with caution -->
-    <plugin.maven-surefire.version>3.1.0</plugin.maven-surefire.version>
+    <plugin.maven-resources.version>3.3.1</plugin.maven-resources.version>
+    <plugin.maven-scm-publish.version>3.3.0</plugin.maven-scm-publish.version>
+    <plugin.maven-site.version>3.21.0</plugin.maven-site.version>
+    <plugin.maven-source.version>3.3.1</plugin.maven-source.version>
+    <plugin.maven-surefire.version>3.5.2</plugin.maven-surefire.version>
```
Comment (cfkoehler) [inline pom.xml:119]: not available yet. (surefire-junit-platform)
have 3.2.3

File: pom.xml [hunk 3]
```diff
@@ -87,41 +87,40 @@
...
     <plugin.maven-replacer.version>1.5.3</plugin.maven-replacer.version>
-    <plugin.maven-resources.version>3.3.0</plugin.maven-resources.version>
-    <plugin.maven-scm-publish.version>3.2.1</plugin.maven-scm-publish.version>
-    <plugin.maven-site.version>3.12.1</plugin.maven-site.version>
-    <plugin.maven-source.version>3.2.1</plugin.maven-source.version>
-    <!-- seems to have an impact with java 11 integration, update with caution -->
-    <plugin.maven-surefire.version>3.1.0</plugin.maven-surefire.version>
+    <plugin.maven-resources.version>3.3.1</plugin.maven-resources.version>
+    <plugin.maven-scm-publish.version>3.3.0</plugin.maven-scm-publish.version>
+    <plugin.maven-site.version>3.21.0</plugin.maven-site.version>
```
Comment (cfkoehler) [inline pom.xml:117]: not available yet

**Review by cfkoehler — COMMENTED** [review]
Not done checking yet.
Ran the following goals so far (compile, verify, site)

**Review by jpdahlke — CHANGES_REQUESTED** [review]
I was able to build these changes internally using this 'incremental' path of commands: 
```
mvn clean verify
mvn clean install
mvn clean verify -Pcheckstyle
mvn clean verify -Pcheckstyle,pmd,coverage
mvn clean site
mvn clean deploy (failed to connect to repo due to access but plugin worked)
mvn clean install site -Pcheckstyle,pmd,coverage
```
All of which 'worked' (success at the end) but I believe I could have just done the last command and been covered (except for deploy), it just would have been harder to find issues with missing dependencies.

I did run into an issue where the maven-project-info-reports had errors when using the cyclonedx plugin. I recommend bumping that to the latest 2.9.1 version as well as adding this plugin dependency to try to manage the version being used with cyclone (it uses 2.0.8 now):
```
<plugin>
  <groupId>io.takari.maven.plugins</groupId>
  <artifactId>takari-lifecycle-plugin</artifactId>
  <version>2.2.0</version>
</plugin>
```
I also see an error during pmd generating the CPD report for a NoSuchMethodError which complains about org.apache.maven.doxia.sink.Sink.verbatim(). I didn't investigate if that was new or where that might have started but it appears as though there are two versions of doxia being included, 2.0.0 and 1.11.1 using:
```
mvn dependency:resolve-plugins
```
Since the result is not a failure of the build but a failure of report generation I'm ok moving forward and resolving in a follow-up pending your feedback.

File: pom.xml [hunk 4]
```diff
@@ -87,40 +87,40 @@
...
     <plugin.maven-site.version>3.12.1</plugin.maven-site.version>
-    <plugin.maven-source.version>3.2.1</plugin.maven-source.version>
-    <!-- seems to have an impact with java 11 integration, update with caution -->
-    <plugin.maven-surefire.version>3.1.0</plugin.maven-surefire.version>
+    <plugin.maven-source.version>3.3.1</plugin.maven-source.version>
+    <plugin.maven-surefire.version>3.2.3</plugin.maven-surefire.version>
     <plugin.nexus-staging-maven.version>1.6.7</plugin.nexus-staging-maven.version>
     <plugin.sonar-maven.version>3.9.1.2184</plugin.sonar-maven.version>
     <plugin.sortpom-maven.version>2.15.0</plugin.sortpom-maven.version>
+    <plugin.takari-lifecycle>2.2.0</plugin.takari-lifecycle>
```
Comment (jpdahlke) [inline pom.xml:123]:
```suggestion
    <plugin.takari-lifecycle>2.0.8</plugin.takari-lifecycle>
```

**Review by jpdahlke — APPROVED** [review]

**Comment by jpdahlke** [comment]
`mvn clean install site -Pcheckstyle,coverage,pmd,dist` worked fully for me

**Review by sambish5 — APPROVED** [review]
works for me!

**Pull request merged**